### DISCUSSION TITLE
20. June 2022

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#667] [Meeting notes]
### Opens

- *Dmitrii* reminded that the Gramine project now keeps Meeting Notes for every week's (Monday) meeting.
- *Don* suggested to add the overall description of the meeting in each Meeting Notes.

### Reorganizing the core repo

Announcement on directory hierarchy & file renames: `LibOS/shim/` -> `libos/`, `Pal/` -> `pal/`, etc.

This reorganization will make the core repository more "sane" and friendly to new Gramine developers. It is a *big* change that will need some getting used to, as well as rebasing all PRs/patches with conflicts created by this reorg.

- Michal and Woju hope that git-rebase will correctly identify file renames, and no conflicts for user PRs/patches will need to be manually fixed.
- There seem to be no big conflicting PRs/patches waiting for merge, so nobody objects to do it now. **It seems the perfect time to do the reorg**.
- The reorganization implies that filenames will be also changed, see below.

1. Replace header-files' guards (`#ifndef _SHIM_HANDLE_H_`) with `#pragma once`
    - Pre-requisite for renaming (because we need to fix guards when renaming e.g. `shim_handle.h` -> `libos_handle.h`)

2. Approved exact renames:
    - [x] `Libos/shim/` -> `libos/`
    - [x] `Pal/` -> `pal/`
    - [x] `Pal/src/host/Linux-SGX/` -> `pal/src/host/linux-sgx/`
    - [x] `Scripts/` -> `scripts/`
    - [x] LibOS files: e.g. `shim_init.c` -> `libos_init.c`
    - [x] `Pal/db_*` -> `Pal/pal_*`
    - [x] `Pal/src/host/Linux-SGX/sgx_*` -> `host_*`
    - [x] `sgx_enclave.*` -> split into `host_ocalls.*` and `host_ecalls.*`
    - [x] `Pal/src/host/Linux-SGX/tools/` -> under root `tools/sgx/`
        - would be good to rename them to e.g. `gramine-sgx-...` but postpone this
 
Rule of thumb: we don't move code around (other than maybe the one change above), postponing it

3. Postponed renames:
    - `LibOS/shim/test/` -> postpone
    - `Pal/regression/` -> postpone
    - All these tests will need serious reorganization: e.g., helper directories should be created on the fly under `/tmp/`.

4. **Not approved** renames:
    - `Pal/src/host/Linux-SGX/` -> `pal-linux-sgx/`

The hope is to finish all this reorg in 1-2 weeks, by the beginning of July 2022.

### Updates on GitHub development

6. Update on PRs, issues and tasks for a previous week:
    - *(no time for this during the meeting, but my take on this is below)*
    - #579 (Big Sockets Rewrite) was finally merged.
    - #650 (Add `O_PATH` support to `open` syscall) was merged, which MariaDB required.
    - #652 (MS Azure Attestation support for RA-TLS) is ready for review.
    - A bunch of small PRs around SGX attestation were submitted and are currently under review.
    - #651 (truncation to zero for Protected Files) is under review and will be merged soon.
    - #609 (support for `getcpu` syscall) is under review and will be merged soon.
    - #654 and #657 track left-overs for SGX attestation and Sockets.

### Agenda for the next meeting

*(please write your proposed agenda items in comments under this discussion)*

- [x] Vijay: SGX EDMM patch
- [x] Dmitrii: IOCTLs PR and the need for a new CI machine (Ubuntu 22.04)

**Comment by svenkata9** [comment]
May I also suggest we create `trusted` and `untrusted` inside `pal-linux-sgx`  directory and separate out the code?

**Reply by dimakuv** [reply]
For the context: the current consensus is on `enclave` (for trusted, enclave part) and `host` (for untrusted, host part).

I am mildly against `trusted` + `untrusted` prefixes. Two reasons:
1. We already have the `enclave` prefix, so we can just re-use it (thus, half of the files do *not* need to be renamed).
2. The `untrusted` prefix is quite long, much longer than the `host` prefix. I like both (they basically imply the same), but the shorter the better.

Anyone else has an opinion on this proposal? @mkow @boryspoplawski @aneessahib ?

**Reply by boryspoplawski** [reply]
1. We do not go with `pal-linux-sgx` as decided on the call
2. I too prefer `enclave`+`host`

**Comment by dimakuv** [comment]
We forgot another item to reorg:

- [x] `Dk` prefix in PAL API functions -> `Pal` (https://github.com/gramineproject/gramine/pull/685)
